models/DataObject/Concrete.php

```diff
@@ -664,6 +664,9 @@ public function save(array $parameters = []): static
 
         try {
             parent::save($parameters);
+            //Reset Relational data to force a reload
+            $this->__rawRelationData = null;
```

Comment:
This line I'd even consider as a bug 🤔 
It seems that `__rawRelationData` is never cleared and it's also not excluded from caching. 
In practice it shouldn't matter, because cache is getting cleared on save, but still, no point including that into serialized data (cache, versioning, ...). 

I'll create a PR for that 😊

Reply:
just for the records #16596 is the PR 😉